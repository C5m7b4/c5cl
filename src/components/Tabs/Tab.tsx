import React from 'react';

export interface TabProps {
  className: string;
  onClick: (e: string) => void;
  title: string;
  activeTab: string;
  disabled: boolean;
  id: string;
}

const Tab = (props: TabProps) => {
  let className = 'tab-list-item';
  if (props.id === props.activeTab) {
    className += ' tab-list-active';
  }

  return (
    <li
      onClick={() => props.onClick(props.id)}
      className={className}
      id={props.id}
    >
      {props.title}
    </li>
  );
};

export default Tab;
