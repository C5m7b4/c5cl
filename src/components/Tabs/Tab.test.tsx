import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Tab from './Tab';

describe('Tab', () => {
  test('should render correctly', () => {
    const testFn = jest.fn();
    const { container } = render(
      <Tab
        id="tab1"
        disabled={false}
        label="Tab1"
        className="tab1"
        onClick={testFn}
        activeTab="tab1"
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should be active tab', () => {
    const testFn = jest.fn();
    const { container } = render(
      <Tab
        id="tab1"
        disabled={false}
        label="tab1"
        className="tb1"
        onClick={testFn}
        activeTab="tab1"
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should handle click', () => {
    const testFn = jest.fn();
    const { container, getByRole } = render(
      <Tab
        id="tab1"
        disabled={false}
        label="tab1"
        className="tb1"
        onClick={testFn}
        activeTab="tab1"
      />
    );
    const li = getByRole('listitem');
    const click = fireEvent.click(li);
  });
});
