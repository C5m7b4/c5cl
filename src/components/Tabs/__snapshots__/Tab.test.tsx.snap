// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should be active tab 1`] = `
<div>
  <li
    class="tab-list-item tab-list-active"
    id="tab1"
  />
</div>
`;

exports[`Tab should render correctly 1`] = `
<div>
  <li
    class="tab-list-item tab-list-active"
    id="tab1"
  />
</div>
`;
